src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -43,6 +43,14 @@ public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContex
                 return StatusCode(StatusCodes.Status400BadRequest, error);
             }
 
+            // Wait for Sidecar specialization to complete before returning ok.
+            // This shouldn't take too long so ok to do this sequentially.
+            error = await _instanceManager.SpecializeMSISidecar(assignmentContext);
```

Comment:
how long does it take on average? and is there any significant cause for variations (e.g: does the MSI container do any external IO during specialization?)

Reply:
700-800 ms. Of this the actual specialization is < 50ms usually. Token service specialization is just updating memory data structures and there is not much variance in the time spent there.

Still need to investigate why the intra-pod communication is slower than expected, but given the expectation that msi adds to cold start , we should be ok with this.